File: module-load-balance-gateway/src/main/java/com/xmmufan/permission/domain/rbac/UserGroup.java
package com.xmmufan.permission.domain.rbac;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.GenericGenerator;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.util.List;

/**
 * @author 黄源钦
 * @Description
 * @Date 11:03
 */
@Entity
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class UserGroup {

    @Id
    @GeneratedValue(generator = "idGenerator")
    @GenericGenerator(name = "idGenerator", strategy = "uuid")
    @Column(length = 64, nullable = false)
    private String id;

    @NotNull
    @Column(unique = true)
    private String userGroupName;

    @NotNull
    private String description;

    @ManyToMany
    @JoinTable(name = "userGroup_permission_mapper")
    private List<Permission> basicPermissions;

    @OneToMany
    @JsonIgnore
    private List<User> users;

}

File: module-load-balance-gateway/src/main/java/com/xmmufan/permission/domain/rbac/UserGroup.kt
package com.xmmufan.permission.domain.rbac

import com.fasterxml.jackson.annotation.JsonIgnore
import org.hibernate.annotations.GenericGenerator
import java.io.Serializable

import javax.persistence.*
import javax.validation.constraints.NotNull

/**
 * @author 黄源钦
 * @Description
 * @Date 11:03
 */
@Entity
data class UserGroup(

        @Id
        @GeneratedValue(generator = "idGenerator")
        @GenericGenerator(name = "idGenerator", strategy = "uuid")
        @Column(length = 64, nullable = false)
        var id: String? = null,

        @NotNull
        @Column(unique = true)
        var userGroupName: String? = null,

        @NotNull
        var description: String? = null,

        @ManyToMany
        @JoinTable(name = "userGroup_permission_mapper")
        var basicPermissions: MutableList<Permission>? = null,

        @OneToMany
        @JsonIgnore
        var users: MutableList<User>? = null

) : Serializable
